### PR TITLE
Improve logging and update remote URLs before fetch

### LINKED ISSUE #6: Needs more logging for debugging
From my limited experience running `git-cache-http-server` in production, I do miss a lot of debugging output to help me analyze problems.

1. `git`'s error messages. Git only returns codes 127 and 128 for every kind of error, so that it's not clear what has actually happened or what the problem is.

2. More details for fetch/clone/update, as to which repository is currently updating, where the fetch or clone goes and also clone/fetch results (`git` output). Also, showing used credentials would be very nice, probably partly obfuscated. 

3. Both fetch and clone return "updating: success" as their completion result, in which case it's not really clear which one succeeded.

4. There should be a severity prefix for the log lines -- "fetch failed, cloning" could be a "WARN", while "updating: success" should be a "INFO". It's easier to scan the logs this way.

### LINKED ISSUE #7: Need to update credentials every time
We are using GitLab CI for our builds. GitLab issues a clone token for every build job, which expires immediately after the build job is finished. `git-cache-http-server` fails to update the credentials on fetch (and apparently also an clone) so that newest revisions are not found in the repository.

Also it seems that re-clone fails (no new revisions in the repository), but doesn't reflect that in the logs:
```
authenticating on the upstream repo
updating: fetching
updating: fetch failed, cloning
promise fulfilled
git-upload-pack done with exit 0
```

Maybe the error handling is flawed at that point.

### PR DESCRIPTION
**Pull request by rassie**
This should fix #6 and #7 in terms of "would work with GitLab CI". One shortcoming is (probably) submodule handling, but we'd get 80% there with this patch alone.

Fixes #6 / Fixes #7

### CODE REVIEW

File: src/Main.hx [hunk 1]
```diff
@@ -48,20 +50,25 @@ class Main {
 	{
 		if (!updatePromises.exists(local)) {
 			updatePromises[local] = new Promise(function(resolve, reject) {
-				trace("updating: fetching");
-				fetch(local, function (ferr, stdout, stderr) {
+				trace('INFO: updating: fetching from $remote');
```
Comment (jonasmalacofilho) [inline src/Main.hx:53]: I think this needs to be filtered a bit: `remote` will contain password/token info.

Note: tokens can appear after the `:` (as a password would), [before the `:`](https://github.com/blog/1270-easier-builds-and-deployments-using-git-over-https-and-oauth), or both.

Comment (jonasmalacofilho) [reply src/Main.hx:53]: Oh, and this is awesome, thanks a lot!

Comment (rassie) [reply src/Main.hx:53]: Hmm, yes, in general, this should be sanitized. In my special use-case, clone tokens are already sanitized by GitLab....  Could something like `params.repo` just work?

Comment (jonasmalacofilho) [reply src/Main.hx:53]: We would also like to log the username, somehow.

Maybe we could create a `infos:{ url:String, user:Null<String> }` value for use in the logs

```haxe
var user = params.auth != null ? parseAuth(params.auth).split(":")[0] : null;
var infos = {
    url : params.repo,
    user : user.length > 5 ? user.substr(0, 5) + "..." : user
}
```

which could later be used in

```haxe
static function update(remote, local, infos:{ url:String, user:String }, callback)
    trace('INFO: updating: fetching from ${infos.url} (${infos.user == null ? "<guest>" : $user})');
```

Leaking 5 bytes of an OAuth token doesn't seem so bad to me...  In fact, it seems enough for some basic auth debugging without significantly impacting the token security.

**Pull request merged**